Docker.md

```diff
@@ -2,8 +2,19 @@
 
 ### Publishing ports
 
-Ports for services should be explicitly published to the loopback interface, 127.0.0.1 to prevent docker from binding them to 0.0.0.0.
-For example, when using postgres or elasticsearch, we might want to make them available to developers, but in the case that we deploy on a public server for testing we don't want them to be exposed.
+By default, containers within the same docker-compose network are able to connect to services running
+on other containers on any port by using the container name, as long as the container is listening
+on 0:0:0:0.
```

Comment:
It is not clear what means “as long as the container is listening on 0:0:0:0.” It doesn’t have to either publish or expose such port to be reachable by another container?

(Besides, it should be `0.0.0.0` instead with dots as separators rather than columns)

Reply:
see my recommended change too - I meant to say here "as long as the _service_ is listening on 0.0.0.0" (yes, dot rather than colon). I mean service as in the program running in the container, not the docker-compose concept of service. Do you think the terminology is OK?

Reply:
Proposed to replace with "the service running in the container" in https://github.com/metabrainz/guidelines/pull/18/commits/f087b462cae59ecf958cf9d03aaacf6c0dce2018 to distinguish from the [service configuration](https://docs.docker.com/compose/compose-file/compose-file-v3/#service-configuration-reference) in a Docker Compose project.